### PR TITLE
Add Europarl-ST SRT recipe with LCMA-SRT

### PR DESCRIPTION
**Pull request by linanjie0820**
## Summary

- Add joint multilingual speech recognition and translation (SRT) recipe for Europarl-ST dataset
- Model: LCMA-SRT ([Language-Conditional Mixture-of-Experts Adapters for Joint Multilingual Speech Recognition and Translation](https://aclanthology.org/2026.acl-long.1634.pdf))
- 9 languages, 72 translation directions
- Includes: data preparation pipeline, 2-stage training scripts, decoding scripts, and pre-trained model links

## Structure

egs/europarl_st/SRT/
├── prepare.sh                     # Data preparation
├── local/                             # Preprocessing scripts
├── lcma_srt/                       # Model code
│   ├── train/stage1/            # ASR pretraining scripts
│   ├── train/stage2/            # Joint SRT training scripts
│   ├── decode/stage1/       # ASR decoding scripts
│   └── decode/stage2/       # Joint decoding scripts
├── README.md
└── RESULTS.md

## Reference

- Paper: https://aclanthology.org/2026.acl-long.1634/
- Pre-trained checkpoints: https://osf.io/rnuhv/
- Standalone repo: https://github.com/linanjie0820/LCMA-SRT

## Summary by CodeRabbit

* **New Features**
  * Added an end-to-end Europarl-ST SRT recipe, including distributed training launchers, a configurable decoding pipeline, and multilingual stage-based decode scripts.
  * Introduced LCMA-SRT model components for ASR/translation, with streaming-capable encoding and optional MoE routing.

* **Documentation**
  * Added full project and recipe documentation, plus static multilingual results tables and checkpoint references.
  * Added detailed data-prep instructions and per-stage script guides.

* **Chores**
  * Added Apache 2.0 licensing text and expanded local preprocessing utilities (manifest checks/filtering, text normalization, audio conversion, FBANK cut generation, and BPE training).

### CODE REVIEW

**Comment by csukuangfj** [comment]
Thank you for your contribution!

We are reviewing it.

**Comment by csukuangfj** [comment]
Please ignore the failed CI tests.

**Comment by linanjie0820** [comment]
> Please ignore the failed CI tests.

Thanks for reviewing.

Looking forward to your feedback.

**Comment by csukuangfj** [comment]
https://osf.io/request-access/rnuhv

Screenshot 2026-07-10 at 10 58 17

The checkpoints currently require permission to access.

Since both the code and the dataset are open-sourced, would it be possible to make the checkpoints publicly available as well?

We recommend hosting the checkpoints on platforms such as ModelScope or Hugging Face.

All recipes in icefall provide publicly accessible checkpoints for reproducibility.

**Comment by linanjie0820** [comment]
> https://osf.io/request-access/rnuhv
> 
> Screenshot 2026-07-10 at 10 58 17
> The checkpoints currently require permission to access.
> 
> Since both the code and the dataset are open-sourced, would it be possible to make the checkpoints publicly available as well?
> 
> We recommend hosting the checkpoints on platforms such as ModelScope or Hugging Face.
> 
> All recipes in icefall provide publicly accessible checkpoints for reproducibility.

Thanks for pointing this out. I’ll make the model checkpoints publicly available on Hugging Face later today and update the repository with the link once the upload is complete.

File: README.md [hunk 1]
```diff
@@ -85,6 +85,136 @@ We would like to highlight the performance of some of the recipes here.
 This is the simplest ASR recipe in `icefall` and can be run on CPU.
 Training takes less than 30 seconds and gives you the following WER:
 
+<<<<<<< Updated upstream
+=======
```
Comment (csukuangfj) [inline README.md]: Can you resolve the merge conflicts?

Comment (linanjie0820) [reply README.md]: I apologize, this was my oversight, it has now been fixed.

File: egs/europarl_st/SRT/lcma_srt/decode/stage1/decode_cr_ctc.sh [hunk 2]
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+set -e
+export DISABLE_VERSION_CHECK=1
+
+echo "=== Training script started on $(hostname) at $(date) ==="
```
Comment (csukuangfj) [inline egs/europarl_st/SRT/lcma_srt/decode/stage1/decode_cr_ctc.sh]:
```suggestion
echo "=== Decoding script started on $(hostname) at $(date) ==="
```

File: egs/europarl_st/SRT/lcma_srt/decode/stage1/decode_cr_ctc_moe.sh [hunk 3]
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+set -e
+export DISABLE_VERSION_CHECK=1
+
+echo "=== Training script started on $(hostname) at $(date) ==="
```
Comment (csukuangfj) [inline egs/europarl_st/SRT/lcma_srt/decode/stage1/decode_cr_ctc_moe.sh]:
```suggestion
echo "=== Decoding script started on $(hostname) at $(date) ==="
```

File: egs/europarl_st/SRT/lcma_srt/decode/stage1/decode_cr_ctc_s_bias.sh [hunk 4]
```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+set -e
+export DISABLE_VERSION_CHECK=1
+
+echo "=== Training script started on $(hostname) at $(date) ==="
```
Comment (csukuangfj) [inline egs/europarl_st/SRT/lcma_srt/decode/stage1/decode_cr_ctc_s_bias.sh]:
```suggestion
echo "=== Decoding script started on $(hostname) at $(date) ==="
```

File: egs/europarl_st/SRT/lcma_srt/decode/stage1/decode_cr_ctc_sc_moe.sh [hunk 5]
```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+set -e
+export DISABLE_VERSION_CHECK=1
+
+echo "=== Training script started on $(hostname) at $(date) ==="
```
Comment (csukuangfj) [inline egs/europarl_st/SRT/lcma_srt/decode/stage1/decode_cr_ctc_sc_moe.sh]:
```suggestion
echo "=== Decoding script started on $(hostname) at $(date) ==="
```

File: egs/europarl_st/SRT/lcma_srt/decode/stage2/decode_hent_srt_m2m.sh [hunk 6]
```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env bash
+set -e
+export DISABLE_VERSION_CHECK=1
+
+echo "=== Training script started on $(hostname) at $(date) ==="
```
Comment (csukuangfj) [inline egs/europarl_st/SRT/lcma_srt/decode/stage2/decode_hent_srt_m2m.sh]:
```suggestion
echo "=== Decoding script started on $(hostname) at $(date) ==="
```

File: egs/europarl_st/SRT/lcma_srt/decode/stage2/decode_hent_srt_m2o.sh [hunk 7]
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+set -e
+export DISABLE_VERSION_CHECK=1
+
+echo "=== Training script started on $(hostname) at $(date) ==="
```
Comment (csukuangfj) [inline egs/europarl_st/SRT/lcma_srt/decode/stage2/decode_hent_srt_m2o.sh]:
```suggestion
echo "=== Decoding script started on $(hostname) at $(date) ==="
```

File: egs/europarl_st/SRT/lcma_srt/decode/stage2/decode_lcma_srt.sh [hunk 8]
```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env bash
+set -e
+export DISABLE_VERSION_CHECK=1
+
+echo "=== Training script started on $(hostname) at $(date) ==="
```
Comment (csukuangfj) [inline egs/europarl_st/SRT/lcma_srt/decode/stage2/decode_lcma_srt.sh]:
```suggestion
echo "=== Decoding script started on $(hostname) at $(date) ==="
```

**Review by csukuangfj — COMMENTED** [review]
Thanks! Looks good to me. 

Left only some minor comments about typo fixes.

I will try to add CI to test the checkpoints after merging it.

**Review by csukuangfj — APPROVED** [review]
Thank you for your contribution!

**Comment by linanjie0820** [comment]
> Thanks! Looks good to me.
> 
> Left only some minor comments about typo fixes.
> 
> I will try to add CI to test the checkpoints after merging it.

Thank you for reviewing the changes and pointing out the spelling error. I have addressed it.

**Comment by linanjie0820** [comment]
> Thank you for your contribution!

Thank you for your review.
